rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -1901,14 +2156,19 @@ static CddsPublisher * create_cdds_publisher(
     rmw_cyclonedds_cpp::make_message_value_type(type_supports));
   struct ddsi_sertype * stact;
   topic = create_topic(dds_ppant, fqtopic_name.c_str(), sertype, &stact);
+
+  dds_listener_t * listener = dds_create_listener(&pub->user_callback_data);
```

Comment:
This allocates a `dds_listener_t` object that is never freed (`dds_create_writer` copies it in, it doesn't take ownership). The same is the case in `create_cdds_subscription`.

Reply:
I see. Would this be the right approach for deleting a listener?:
```
dds_listener_t * listener = dds_create_listener(&sub->user_callback_data);
sub->enth = dds_create_reader(dds_sub, topic, qos, listener);
dds_delete_listener(listener);
```
That is, freeing the listener once the reader has been created.

Reply:
Yep!

The reason you have to allocate/free `dds_listener_t` (and the same for `dds_qos_t`) is so we can add things without breaking binary compatibility. That happens so rarely that I am not sure it is worth it ... but that's how it is for now.